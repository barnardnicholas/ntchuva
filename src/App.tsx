import React, { useState } from 'react';
import Game from './components/Game';
import Settings from './components/settings/Settings';
import Header from './components/header/Header';
import './_styles/App.scss';
import BackgroundBlocker from './components/BackgroundBlocker';

function App() {
  const [showSettings, setShowSettings] = useState<boolean>(false);

  return (
    <div className="App">
      <Game />
      <BackgroundBlocker showSettings={showSettings} handleClick={() => setShowSettings(false)} />
      <Settings showSettings={showSettings} />
      <Header showSettings={showSettings} setShowSettings={setShowSettings} />
    </div>
  );
}

export default App;
