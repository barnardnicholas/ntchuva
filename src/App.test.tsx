import React from 'react';
import renderer from 'react-test-renderer';
// import { render, screen } from '@testing-library/react';
import App from './App';

it('Renders the App correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
