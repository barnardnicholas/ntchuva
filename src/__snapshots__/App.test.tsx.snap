// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the App correctly 1`] = `
<div
  className="App"
>
  <div
    className="app-content"
  >
    <header>
      <h1>
        Ntchuva
      </h1>
      <div>
        Bao la Kujifunza/Bao for Beginners/Mbili-mbili
      </div>
    </header>
    <main>
      <div>
        Player 1's turn
      </div>
      <div
        style={
          Object {
            "height": "1rem",
          }
        }
      />
      <div
        className="board-container"
      >
        <div
          className="board "
          id="board1"
        >
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "51.0%",
                  "top": "39.9%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "29.0%",
                  "top": "40.1%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "33.2%",
                  "top": "31.4%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "46.8%",
                  "top": "48.6%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "43.3%",
                  "top": "29.5%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "36.7%",
                  "top": "50.5%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "46.3%",
                  "top": "31.0%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "33.7%",
                  "top": "49.0%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "50.9%",
                  "top": "38.3%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "29.1%",
                  "top": "41.7%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "43.4%",
                  "top": "29.5%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "36.6%",
                  "top": "50.5%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "32.8%",
                  "top": "31.7%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "47.2%",
                  "top": "48.3%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "29.2%",
                  "top": "37.9%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "50.8%",
                  "top": "42.1%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "49.9%",
                  "top": "35.3%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "30.1%",
                  "top": "44.7%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "29.0%",
                  "top": "39.4%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "51.0%",
                  "top": "40.6%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "47.5%",
                  "top": "31.9%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "32.5%",
                  "top": "48.1%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "40.9%",
                  "top": "29.0%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "39.1%",
                  "top": "51.0%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "31.2%",
                  "top": "33.4%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "48.8%",
                  "top": "46.6%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "50.7%",
                  "top": "37.4%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "29.3%",
                  "top": "42.6%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "36.2%",
                  "top": "29.7%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "43.8%",
                  "top": "50.3%",
                }
              }
            />
          </div>
          <div
            className="board-square  disabled"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "34.4%",
                  "top": "30.5%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "45.6%",
                  "top": "49.5%",
                }
              }
            />
          </div>
          <div
            className="anim-hand-widget "
            style={
              Object {
                "left": "calc(50% - 1rem)",
                "opacity": 0,
                "top": "calc(50% - 1rem)",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "height": "1rem",
            }
          }
        />
        <div
          className="board active"
          id="board0"
        >
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "50.9%",
                  "top": "38.7%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "29.1%",
                  "top": "41.3%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "50.2%",
                  "top": "35.9%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "29.8%",
                  "top": "44.1%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "49.4%",
                  "top": "34.2%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "30.6%",
                  "top": "45.8%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "40.9%",
                  "top": "29.0%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "39.1%",
                  "top": "51.0%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "50.9%",
                  "top": "38.6%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "29.1%",
                  "top": "41.4%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "29.2%",
                  "top": "37.7%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "50.8%",
                  "top": "42.3%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "33.8%",
                  "top": "30.9%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "46.2%",
                  "top": "49.1%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "47.3%",
                  "top": "31.8%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "32.7%",
                  "top": "48.2%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "50.0%",
                  "top": "35.4%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "30.0%",
                  "top": "44.6%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "34.6%",
                  "top": "30.4%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "45.4%",
                  "top": "49.6%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "33.2%",
                  "top": "31.4%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "46.8%",
                  "top": "48.6%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "49.3%",
                  "top": "34.1%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "30.7%",
                  "top": "45.9%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "44.8%",
                  "top": "30.1%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "35.2%",
                  "top": "49.9%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "32.6%",
                  "top": "31.9%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "47.4%",
                  "top": "48.1%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "51.0%",
                  "top": "39.6%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "29.0%",
                  "top": "40.4%",
                }
              }
            />
          </div>
          <div
            className="board-square  "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="counter"
              style={
                Object {
                  "left": "29.6%",
                  "top": "36.3%",
                }
              }
            />
            <div
              className="counter"
              style={
                Object {
                  "left": "50.4%",
                  "top": "43.7%",
                }
              }
            />
          </div>
          <div
            className="anim-hand-widget "
            style={
              Object {
                "left": "calc(50% - 1rem)",
                "opacity": 0,
                "top": "calc(50% - 1rem)",
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "height": "1rem",
          }
        }
      />
      <div
        className="button-container"
      >
        <span>
          P1: 
          32
        </span>
        <span>
          P2: 
          32
        </span>
      </div>
    </main>
    <footer />
  </div>
</div>
`;
